runtime/lua/vim/lsp/util.lua

```diff
@@ -568,10 +569,16 @@ do
 
   function M.buf_clear_diagnostics(bufnr)
     validate { bufnr = {bufnr, 'n', true} }
-    bufnr = bufnr == 0 and api.nvim_get_current_buf() or bufnr
+    bufnr = api.nvim_get_current_buf() or bufnr
```

Comment:
sorry, to clarify: this assignment isn't needed at all. `0` already means "current buffer"  so `nvim_get_current_buf` was never necessary.

Reply:
sorry, my bad.  So if this function receives `nil`, it will be passed to `nvim_buf_clear_namespace`. Does this function handle `nil`?

Reply:
the old code didn't handle `nil` so that's a different topic...